File: infrastructures/ddd/csharp/src/Infrastructure/Repositories/IExampleCommandRepository.cs
using OddlyDdd.Domain.Models;
using OddlyDdd.Infrastructure.Repositories.Infra;

namespace OddlyDdd.Infrastructure.Repositories
{
    /// <summary>
    /// Example command repository interface demonstrating CQRS command side.
    /// Command repositories:
    /// - Work with WriteEntities internally
    /// - Receive BMOs (Business Model Objects) from services
    /// - Return void or IDs
    /// - Handle mapping BMO ↔ WriteEntity internally using mapper
    /// - Used for: Create, Update, Delete operations
    /// </summary>
    public interface IExampleCommandRepository : ICommandRepository<ExampleModel, string>
    {
        // Additional command-specific methods can be added here if needed
        // Base interface provides: SaveAsync, UpdateAsync, DeleteAsync, ExistsAsync
    }
}

File: infrastructures/ddd/java/src/main/java/com/oddly/ddd/infrastructure/repositories/IExampleCommandRepository.java
package com.oddly.ddd.infrastructure.repositories;

import com.oddly.ddd.domain.models.ExampleModel;
import com.oddly.ddd.infrastructure.repositories.infra.ICommandRepository;

/**
 * Command repository interface for Example domain.
 * Command repositories:
 * - Live in /infrastructure/repositories/
 * - Have ICommandRepository suffix
 * - Implementations go in /infrastructure/repositories/impl/
 * - Work with WriteEntities and receive BMOs
 * - Handle mapping from BMO to WriteEntity internally
 */
public interface IExampleCommandRepository extends ICommandRepository<ExampleModel, String> {
    // Additional command-specific methods for Example can be added here
}
